### DISCUSSION TITLE
Migrating our container images to GitHub Container Registry

### DISCUSSION THREAD

**Discussion by tomkerkhove** [kedacore/keda#1700] [Deprecations]
We provide **various ways to [deploy KEDA](https://keda.sh/docs/latest/deploy/) in your cluster** including by using [Helm chart](https://github.com/kedacore/charts), [Operator Hub](https://operatorhub.io/operator/keda) and raw YAML specifications.

These deployment options all rely on the container images that we provide which are available on **[Docker Hub](https://hub.docker.com/u/kedacore), the industry standard for public container images**.

However, we have found that Docker Hub is no longer the best place for our container images and are migrating to GitHub Container Registry (Preview).

## What is changing?

Our container images are being published on [GitHub Container Registry](https://github.com/orgs/kedacore/packages?type=source) for end-users to pull them.

Because of this, the names of our container images are changing:

| Component      | New Image (GitHub Container Registry)     | Legacy Image (Docker Hub)         |
| :------------- | :---------------------------------------- | --------------------------------- |
| Metrics Server | `ghcr.io/kedacore/keda-metrics-apiserver` | `kedacore/keda-metrics-apiserver` |
| Operator       | `ghcr.io/kedacore/keda`                   | `kedacore/keda`                   |

## When is this taking place?

As of v2.2, we have started publishing our new container images to GitHub Container Registry in parallel to Docker Hub.

This allows customers to already migrate to our new registry and consume our artifacts there.

**Once GitHub Container Registry becomes generally available (GA), we will no longer publish new versions to Docker Hub.**

## What is the impact for end-users?

**If you are using one of our deployment options, end-users are not be impacted.**

Since v2.2, we are using GitHub Container Registry by default and you are good to go.

If you are using your own deployment mechanism, then you will have to pull the container images from GitHub Container Registry instead.

## Learn more

Learn more about why we are doing this in [our blog post](https://keda.sh/blog/migrating-to-github-container-registry/) or have a look at our [open issue](https://github.com/kedacore/governance/issues/16).

**Comment by benc-uk** [comment]
Good plan!
